streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPeek.java

```diff
@@ -35,10 +48,39 @@ public KStreamPeek(final ForeachAction<K, V> action) {
     }
 
     private class KStreamPeekProcessor extends AbstractProcessor<K, V> {
+        
+        @Override
+        public void init(ProcessorContext context) {
+            super.init(context);
+            if (print) {
+                printAction.setContext(context);
+                if (printAction.keySerde() == null) {
+                    printAction.useDefaultKeySerde();
+                }
+                if (printAction.valueSerde() == null) {
+                    printAction.useDefaultValueSerde();
+                }
```

Comment:
Using 5 methods seems rather complex. I think we can move all this logic into a single method of `PrintForeachAction`